autoload/ale/hover.vim

```diff
@@ -24,7 +24,13 @@ function! ale#hover#HandleTSServerResponse(conn_id, response) abort
 
         if get(a:response, 'success', v:false) is v:true
         \&& get(a:response, 'body', v:null) isnot v:null
-            call ale#util#ShowMessage(a:response.body.displayString)
+            if get(l:options, 'hover_from_balloonexpr', 0)
+            \&& exists('*balloon_show')
+            \&& ale#Var(l:options.buffer, 'set_balloons')
```

Comment:
We'll need to think about the meaning of the `set_balloons` setting. I guess it makes sense how you've done it here. I doubt many people really look at the balloons for the error messages at the moment.

I say we should do it how you've done it here, and update the documentation to say that the option also enables hover information for LSP and tsserver.

I might update the tsserver stuff myself, or you can give it a go if you want. If you install TypeScript into `node_modules` for a project, it should be pretty easy to get it running `tsserver` comes with TypeScript.

Reply:
I assumed that the `ale#hover` callbacks were only called by `ale#hover#Show`, and therefore the value of set_ballons in this context is only relevant for the Hover information to begin with. I think this specific part of the code is not related to the actual design decision, it's really just a sanity check on what the user asked for relatively to balloons before we make it appear.

The part that makes the meaning of `set_balloons` matter in a "hover vs diagnostics" way is in `ale#balloon#MessageForPos` (where I decided that if diagnostics are found, show them and don't even try to get hover information for the position).